src/bp/ui-studio/src/web/views/FlowBuilder/diagram/manager.ts

```diff
@@ -119,8 +119,24 @@ export class DiagramManager {
     this.getLinksRequiringUpdate()
   }
 
-  shouldHighlightNode(nodeName): boolean {
-    return this.highlightedNodeNames && !!this.highlightedNodeNames.find(x => nodeName.includes(x))
+  shouldHighlightNode(node: NodeView): boolean {
+    if (_.isEmpty(this.highlightedNodeNames)) {
+      return false
+    }
+
+    return this.highlightedNodeNames
+      .map(q => {
+        // quick and dirty way to filter on node content
+        const candidates = [
+          node.name,
+          node.id,
+          JSON.stringify(node.contents || {}),
+          JSON.stringify(node.prompt?.params.question ?? {}),
+          JSON.stringify(node.conditions || {})
+        ]
+        return [q, candidates] as [string, string[]]
```

Comment:
A lot happening here (map then includes inside some inside some)

why not:

```
return this.highlightedNodeNames.some(q => {

const candidates = [
          node.name,
          node.id,
          JSON.stringify(node.contents || {}),
          JSON.stringify(node.prompt?.params.question ?? {}),
          JSON.stringify(node.conditions || {})
        ]

const searchCorpus = candidates.join().toLowerCase()
return searchCorpus.includes(q.toLowerCase()
})
```

Reply:
I came up with this because I wanted this to be clean and readable.. if you don't like it then I can use my previous implementation which involves a magical for loop with a massive one liner ><